### PR TITLE
Minor changes and bugfixes for Quantity.

### LINKED ISSUE #235: Minor updates and bugfixes for Quantity
Quantity is not rounded to 3 d.p. for discrete ingredients (without units).
Quantity is stored as a String, should change to a double and a String.

### PR DESCRIPTION
**Pull request by NayLin-H99**
Fix #235

### CODE REVIEW

File: src/main/java/fridgy/model/ingredient/Quantity.java [hunk 1]
```diff
@@ -27,20 +28,28 @@
...
-        AppUtil.checkArgument(isValidQuantity(quantity), MESSAGE_CONSTRAINTS);
-        value = convertToStandardUnit(quantity);
+        AppUtil.checkArgument(isValidQuantityString(quantity), MESSAGE_CONSTRAINTS);
+        String[] results = QuantityCalc.standardiseQuantity(quantity).split(" ");
+        value = Double.parseDouble(results[0]);
+        if (results.length > 1) {
+            units = results[1];
+        } else {
+            units = "";
+        }
```
Comment (rootkie) [inline src/main/java/fridgy/model/ingredient/Quantity.java:38]: Hide this parsing through splitting string in a method, keep constructor easy to read.
```suggestion
        String standardQuantity = QuantityCalc.standardiseQuantity(quantity);
        this.value = getQuantityValue(standardQuantity);
        this.units = getQuantityUnits(standardQuantity);
```

File: src/main/java/fridgy/model/ingredient/Quantity.java [hunk 2]
```diff
@@ -56,25 +65,23 @@ public int hashCode() {
     }
 
     private String convertToStandardUnit(String quantity) {
-        return QuantityCalc.convertToStandardUnits(quantity);
+        return QuantityCalc.standardiseQuantity(quantity);
     }
```
Comment (rootkie) [inline src/main/java/fridgy/model/ingredient/Quantity.java:69]: is this method still needed?

File: src/main/java/fridgy/model/ingredient/Quantity.java (hunk 1 above)
Comment (NayLin-H99) [reply src/main/java/fridgy/model/ingredient/Quantity.java]: okay adjusted 🐒

File: src/main/java/fridgy/model/ingredient/Quantity.java (hunk 2 above)
Comment (NayLin-H99) [reply src/main/java/fridgy/model/ingredient/Quantity.java]: ok adjusted 🐒

**Review by rootkie — APPROVED** [review]
LGTM

**Pull request merged**